src/peerDiscovery.cpp

```diff
@@ -0,0 +1,378 @@
+                           &config_ipv6, 
+                           sizeof(config_ipv6)) < 0
+            ){
+                throw std::runtime_error(std::string("setsockopt_mcast_join") + strerror(errno));
+            }
+
+            break;
+
+    }       
+
+}
+
+void
+PeerDiscovery::m_sendto(uint8_t *buf,size_t &buf_size){
+
+    int nbytes = sendto(
+        m_sockfd,
+        buf,
+        buf_size,
+        0,
+        m_sockaddr.get(),
+        m_sockaddr.getLength()
+    );
+    if (nbytes < 0) {
+
+        throw std::runtime_error(std::string("sendto_m_sendto") + strerror(errno));
+
+    }
+
+}
+
+void
+PeerDiscovery::m_recvfrom(uint8_t *buf,size_t &buf_size)
+{
+    socklen_t sa_len = sizeof(m_sockaddr);
+    int nbytes = recvfrom(
+        m_sockfd,
+        buf,
+        buf_size,
+        0,
+        m_sockaddr.get(),
+        &sa_len
+    );
+    if (nbytes < 0) {
+        throw std::runtime_error(std::string("recvfrom_m_recvfrom") + strerror(errno));
+    }
+
+}
+
+void
+PeerDiscovery::Sender_oneTimeShoot(dht::InfoHash nodeId, in_port_t port_to_send){
+
+    //Set up for Sender
+    initialize_socket();
+    initialize_sockaddr_Sender();
+
+    //Setup for send data
+    uint8_t data_send[22];
+    uint8_t *x = nodeId.data();
+
+    size_t data_send_size = 22;
+    int port_node = port_to_send;
+    uint8_t port_node_binary[2];
+    PeerDiscovery::inttolitend(port_node,port_node_binary);
+
+    //Copy Node id and node port
+    memcpy (data_send, x, 20);
+    data_send[20] = port_node_binary[0];
+    data_send[21] = port_node_binary[1];
+
+    m_sendto(data_send,data_send_size);
+
+}
+
+void
+PeerDiscovery::Listener_oneTimeShoot(){
+
+    initialize_socket();
+    initialize_socketopt();
+    initialize_sockaddr_Listener();
+    mcast_join();
+
+    size_t data_receive_size = 22;
+    uint8_t data_receive[22];
+    m_recvfrom(data_receive,data_receive_size);
+
+    uint8_t data_infohash[20];
+    uint8_t data_port[2];
+
+    memcpy (data_infohash, data_receive, 20);
+    data_port[0] = data_receive[20];
+    data_port[1] = data_receive[21];
+
+    m_port_received = PeerDiscovery::litendtoint(data_port);
+    m_node_id_received = dht::InfoHash(data_infohash, 20);
+
+}
+
+void
+PeerDiscovery::Sender_Setup(dht::InfoHash nodeId, in_port_t port_to_send){
+
+    //Set up for Sender
+    initialize_socket();
+    initialize_sockaddr_Sender();
+
+    //Setup for send data
+    uint8_t *x = nodeId.data();
+    int port_node = port_to_send;
+    uint8_t port_node_binary[2];
+    PeerDiscovery::inttolitend(port_node,port_node_binary);
+
+    //Copy Node id and node port
+    memcpy (m_data_send, x, 20);
+    m_data_send[20] = port_node_binary[0];
+    m_data_send[21] = port_node_binary[1];
+    
+}
+
+void
+PeerDiscovery::SenderThread(bool &continues){
+
+    while(continues){
+        
+        m_sendto(m_data_send,m_data_size);
+        sleep(5);
+
+    }
+
+}
+
+void
+PeerDiscovery::Sender_Loop(){
```

Comment:
dedicated method not needed for this. Thread can be instantiated in SenderThread or in startPublish